projects/ngneat/hotkeys/src/lib/hotkeys.service.ts

```diff
@@ -100,9 +105,10 @@ export class HotkeysService {
 
       return () => {
         this.hotkeys.delete(normalizedKeys);
+        this.disposers.delete(normalizedKeys);
         dispose();
       };
-    });
+    }).pipe(takeUntil<KeyboardEvent>(disposer));
```

Comment:
I'd create one general subject that emits the normalizedKeys upon destroy:

```ts
dispose = new Subjet<string>();


.pipe(takeUntil<KeyboardEvent>(this.dispose.pipe(filter(v => v === normalizedKeys))))



removeShortcuts() {
  this.dispose.next(hotkey)
}
```

Reply:
Okay, that's a good idea.
But I think we still need `this.hotkeys.delete(hotkey);`  in removeShortcuts() in the case like below, is it okay?

```
ngOnInit() {
    this.hotkeys.addShortcut('meta.a');
    this.hotkeys.removeShortcuts('meta.a');
    this.hotkeys.addShortcut('meta.a'); // may throw "Duplicated shortcut" error.
}

```

If addShortcut() is called but not subscribed, just hotkeys are added.
And then removeShortcuts() is called, unsubscribe process does not fire because there's any subscription.

This use case is practically does nothing and very rare, but I think it needs to be handled.

Reply:
I have updated it as per your suggestion. @NetanelBasal